### PR TITLE
Add Togls::TestToggleRegistry class

### PR DESCRIPTION
**Pull request by cyphactor**
Why you made the change:

I added this class specifically so that we would have a toggle registry
that didn't use environment override drivers. The is especially
important in the automated test suite scenario as identified by
issue #24. It basically prevents environment variables from interfering
with the explicitly configured toggle states in your automated tests.

How the change addresses the need:

It does so by allowing you in your test setup to set the toggle registry
to an instance of the new Togls::TestToggleRegistry class.

### CODE REVIEW

File: lib/togls/test_toggle_registry.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Togls
+  class TestToggleRegistry < FeatureToggleRegistry
```
Comment (RyanHedges) [inline lib/togls/test_toggle_registry.rb]: Maybe call this `InMemoryToggleRegistry` or, `LocalToggleRegistry`.  There isn't anything in this object related to tests other than the name.

Comment (cyphactor) [inline lib/togls/test_toggle_registry.rb]: I think that takes away from the intention of the use case. The only reason the in memory only registry exists is for the use case of automated tests.

Right now I see three different types `TestToggleRegistry`, `ReleaseToggleRegistry` and `BusinessToggleRegistry`. Currently the `FeatureToggleRegistry` playing the role of the `ReleaseToggleRegistry`.

File: spec/lib/togls/test_toggle_registry_spec.rb [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      block = Proc.new {}
+      expect(registry.expand(&block)).to eq(registry)
+    end
+  end
+
+  describe "#feature" do
+    it "creates a new feature object with the passed key" do
+      desc = double('feature desc')
+      key = "some_key"
+      feature = Togls::Feature.new('your_mom', "Your Moms Desc")
```
Comment (RyanHedges) [inline spec/lib/togls/test_toggle_registry_spec.rb]: Let's change this feature key and description

**Comment by RyanHedges** [comment]
That's all my comments.

**Comment by RyanHedges** [comment]
:+1:

**Pull request merged**